app/controllers/merchant/dashboard_controller.rb

```diff
@@ -1,6 +1,10 @@
 class Merchant::DashboardController < Merchant::BaseController
   def index
     @merchant = Merchant.find(current_user.merchant_id)
-    # Order.joins(:item).where(item: {merchant: @merchant})
+  end
+
+  def show
+    @order = Order.find(params[:id])
+    @merchant_items = @order.items.where("merchant_id =?", current_user.merchant_id)
```

Comment:
Good query here and I don't think this can be in the model since we are calling on current_user